### PR TITLE
Homework 6 UniquePtr class

### PR DESCRIPTION
**Pull request by Ionel14**

### CODE REVIEW

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]
merging against wrong branch

File: uniqueptr.h [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    void swap( UniquePtr& other ) noexcept
+    {
+        if(ptr_ != other.ptr_)
+        {
+            T* auxPtr = this->release();
+            ptr_ = other.release();
+            other.ptr_ = auxPtr;
+
+            auxPtr = nullptr;
+            delete auxPtr;
```
Comment (CristianIonutIonescu) [inline uniqueptr.h]: this part of code doesn't look just right to me, you're deleting a nullptr? why?

File: main.cpp [hunk 2]
```diff
@@ -32,16 +36,46 @@ int main()
 
         // smartHome.AddToXml("../data.xml");
 
-        smart_home::LightSensor light("Light", 600);
-        smart_home::AcUnit ac("AC");
+        smart_home::UniquePtr<int> ptr1(new int(1));
+        smart_home::UniquePtr<int> ptr2(new int(3));
+        
+        smart_home::UniquePtr<int> arr(new int[3]{1,2,3});
```
Comment (CristianIonutIonescu) [inline main.cpp:42]: can you add a new value here:
`smart_home::UniquePtr<int> ptr3 = std::nullptr`
and tell me if it works?

**Review by CristianIonutIonescu — COMMENTED** [review]
looks great, please check the 2 comments I've left

File: uniqueptr.h (hunk 1 above)
Comment (Ionel14) [reply uniqueptr.h]: Correct, I should delete that line

File: main.cpp (hunk 2 above)
Comment (Ionel14) [reply main.cpp:42]: I did it and it works

**Review by CristianIonutIonescu — APPROVED** [review]

**Pull request merged**